pkg/skaffold/runner/new.go

```diff
@@ -43,12 +44,14 @@ import (
 
 // NewForConfig returns a new SkaffoldRunner for a SkaffoldConfig
 func NewForConfig(runCtx *runcontext.RunContext) (*SkaffoldRunner, error) {
+	docker := docker.NewAPI(runCtx)
```

Comment:
This now happens irrespective of the what builder you are using. 
e.g. if i am skaffold user who does not have docker installed locally and all my builds are running on my cluster shd this be still initialized?

Would this not cause an error here https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/docker/client.go#L79 for skaffold user who don't have docker installed?